packages/flutter/lib/src/widgets/notification_listener.dart

```diff
@@ -132,6 +132,34 @@ class NotificationListener<T extends Notification> extends StatelessWidget {
   Widget build(BuildContext context) => child;
 }
 
+typedef Widget NotificationChangedBuilder<T extends Notification>(BuildContext context, Widget child, T notification);
+
+class NotificationBuilder<T extends Notification> extends StatefulWidget {
+  NotificationBuilder({this.builder, this.intercepting: false, this.child});
+
+  final NotificationChangedBuilder<T> builder;
+  final bool intercepting;
+  final Widget child;
+
+  @override
+  State<StatefulWidget> createState() => NotificationBuilderState<T>();
+}
+
+class NotificationBuilderState<T extends Notification> extends State<NotificationBuilder<T>> {
+  T lastNotification;
```

Comment:
maybe a list? there could be many notifications.

Reply:
what does it mean concretely? Does scrollables send multiple notifications per frame?

Reply:
They can, sure. There's nothing about the notification system that limits notifications to one per frame.